src/main/resources/static/user.js

```diff
@@ -74,7 +89,52 @@ function generateSurveyForm(data) {
             </div>
         </div>
     </form>`;
-    
+
     return htmlString;
   });
 }
+
+function addFormListener() {
+  $("#myForm").submit(function (e) {
+    //prevent Default functionality
+    e.preventDefault();
+
+    let formData = $("#myForm").serializeArray();
+
+    console.log(formData);
+    console.log(questionIdTypePair);
+
+    let error = false;
+    let success = true;
+    formData.forEach((data) => {
+      let question_type = questionIdTypePair.get(parseInt(data.name));
+      console.log(question_type);
+      handlePatchRequest(`/survey/${question_type}/${data.name}/submit`, data.value);
+      if (error) {
+        alert("Submit failed");
+      } else {
+        alert("Submitted Successfully!");
+        window.location.href = "/";
```

Comment:
Move this logic to the handle post request function. You don't need the error variable. It can be alerted if it submitted successfully in the done and submit failed in the fail